operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -432,6 +449,66 @@ void reconcileIngressControllers() {
         }
     }
 
+    public void ensureBlueprintRouteMatching(Route route, String blueprintBaseName) {
+        var annotations = Optional.ofNullable(route.getMetadata()).map(ObjectMeta::getAnnotations);
+        var tslConfig = Optional.ofNullable(route.getSpec()).map(RouteSpec::getTls);
+
+        ensureBlueprintRouteMatching(annotations, tslConfig, blueprintBaseName);
+    }
+
+    public void ensureBlueprintRouteMatching(Optional<Map<String, String>> annotations, Optional<TLSConfig> tlsConfig, String blueprintBaseName) {
+        // see findMatchingBlueprint https://github.com/openshift/router/blob/master/pkg/router/template/configmanager/haproxy/manager.go#L817
+        if (!Boolean.TRUE.equals(dynamicConfigManager) || blueprintRouteNamespace == null) {
+            return;
+        }
+
+        var orderedAnnotations = annotations.map(TreeMap::new);
+        MessageDigest stableIdDigest;
+        try {
+            stableIdDigest = MessageDigest.getInstance("SHA-1");
+            orderedAnnotations.ifPresent(m -> m.forEach((k, v) -> {
+                stableIdDigest.update(String.valueOf(k).getBytes());
+                stableIdDigest.update(String.valueOf(v).getBytes());
+            }));
```

Comment:
Spot bugs objecting to this:

```
[INFO] --- spotbugs-maven-plugin:4.2.0:check (basepom.default) @ kas-fleetshard-operator ---
[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
Error:  High: Found reliance on default encoding in org.bf2.operator.managers.IngressControllerManager.lambda$ensureBlueprintRouteMatching$17(MessageDigest, String, String): String.getBytes() [org.bf2.operator.managers.IngressControllerManager, org.bf2.operator.managers.IngressControllerManager] At IngressControllerManager.java:[line 470]Another occurrence at IngressControllerManager.java:[line 471] DM_DEFAULT_ENCODING
```

```suggestion
                stableIdDigest.update(String.valueOf(k).getBytes(StandardCharsets.UTF_8));
                stableIdDigest.update(String.valueOf(v).getBytes(StandardCharsets.UTF_8));
            }));
```
Thankfully its not bothered by the method referenes on lines 476-479